### PR TITLE
Check if a queue exists before trying to create it

### PR DESCRIPTION
**Pull request by rphillipsz**
https://jira.zenoss.com/browse/ZEN-28918

If the arguments used to create the queue are different from the arguments zenoss-protocols is trying to create it as, the create will fail and we can't connect to the queue.  If the queue exists, use the queue as-is without trying to create it.

### CODE REVIEW

**Comment by jplouis** [comment]
Jenkins cannot run this build automatically. Can one of the admins verify this patch?  To kick off the pull verification add a comment with "ok to test".

**Comment by jplouis** [comment]
Build finished. 10 tests run, 0 skipped, 0 failed.

File: python/zenoss/protocols/twisted/amqp.py [hunk 1]
```diff
@@ -144,9 +148,19 @@ def get_queue(self, queue, exclusive=False):
 
     @inlineCallbacks
     def create_queue(self, queue):
+        # Check to see if the queue exists.
+        try:
+            self._checking = True
+            yield getAdapter(self.chan, IAMQPChannelAdapter).declareQueue(queue, True)
```
Comment (jplouis) [inline python/zenoss/protocols/twisted/amqp.py]: Shouldn't _checking be set to false after the check? perhaps in a finally block?

Comment (rphillipsz) [reply python/zenoss/protocols/twisted/amqp.py]: It should, yes.  I'm not at a computer for awhile to make the change but I'll get that added.

**Comment by jplouis** [comment]
Build finished. 10 tests run, 0 skipped, 0 failed.

**Pull request merged**